### PR TITLE
Add some tests covering the application of duplicate Ethereum events/vote extensions

### PR DESCRIPTION
**Pull request by james-chf**
This PR adds test cases to cover a couple of scenarios discussed where a validator could attempt to vote twice on the same event:
- by including the same Ethereum event twice in a single vote extension
- by submitting multiple vote extensions for the same block height, each vote extension containing a same Ethereum event

### CODE REVIEW

File: ethereum_bridge/Cargo.toml [hunk 1]
```diff
@@ -26,6 +26,8 @@ abciplus = [
     "namada_proof_of_stake/abciplus",
 ]
 
+testing = []
```
Comment (james-chf) [inline ethereum_bridge/Cargo.toml:29]: Added a `testing` feature to `namada_ethereum_bridge` crate for guarding test helper functionality (this is not a great way of doing it, but follows the pattern we're currently using - a dedicated test utils crate would be better)

File: ethereum_bridge/src/protocol/transactions/ethereum_events/mod.rs [hunk 2]
```diff
@@ -167,15 +167,47 @@ where
...
+    use namada_core::ledger::storage::testing::TestStorage;
+    use namada_core::ledger::storage::{Sha256Hasher, Storage};
+    use namada_core::types::address::Address;
+    use namada_proof_of_stake::epoched::Epoched;
+    use namada_proof_of_stake::types::{ValidatorSet, WeightedValidator};
+    use namada_proof_of_stake::PosBase;
+
+    /// Set up a `TestStorage` initialized at genesis with validators of equal
+    /// power.
+    pub fn setup_storage(
```
[the remark this answers is not in the document]
Comment (james-chf) [reply ethereum_bridge/src/protocol/transactions/ethereum_events/mod.rs:184]: (this helper is actually more generically useful than the Ethereum bridge so could actually move to a more shared crate/test utils crate)

**Review by sug0 — APPROVED** [review]

**Review by batconjurer — APPROVED** [review]

**Comment by james-chf** [comment]
I need to make some changes to consolidate with the move of the `setup_storage` test helper function which already happened in `eth-bridge-integration`, then will merge this
